src/txn/txn_recover.c

```diff
@@ -503,6 +504,44 @@ __recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
     return (ret);
 }
 
+/*
+ * __recovery_txn_setup_initial_state --
+ *     Setup the transaction initial state required by rollback to stable.
+ */
+static int
+__recovery_txn_setup_initial_state(WT_SESSION_IMPL *session, WT_RECOVERY *r)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_RET(__recovery_set_checkpoint_snapshot(session));
+
+    /*
+     * Set the checkpoint timestamp and oldest timestamp retrieved from the checkpoint metadata.
+     * These are the stable timestamp and oldest timestamps of the last successful checkpoint.
+     */
+    WT_RET(__recovery_set_checkpoint_timestamp(r));
+    WT_RET(__recovery_set_oldest_timestamp(r));
+
+    /*
+     * Now that timestamps extracted from the checkpoint metadata have been configured, configure
+     * the pinned timestamp.
+     */
+    __wt_txn_update_pinned_timestamp(session, true);
+
+    WT_ASSERT(session,
```

Comment:
Is this going to cause problems for read-only connections now? I see that the checks used to only happen for non read-only connections.

Reply:
No. It doesn't cause any problems for read-only connections. We are validating that there is no existing stable timestamp already as we are going to replace it with the stable timestamp from the checkpoint.

Reply:
I think I see - setting the stable timestamp isn't necessary for read-only connections, but also isn't harmful. It shouldn't be used, since recovery will be skipped and new checkpoints can't be created.